Azure-Samples/chat-with-your-data-solution-accelerator
# Disable app insights when running locally

Closes #634 

## Purpose

* Disable app insights when running locally, by using env var `APPINSIGHTS_ENABLED` which is only set in Azure
* Update `TokenLogger` to  not use deprecated opencensus library
* Fix error with custom dimensions
  * Note that the log message is not visible, as Flask sets the default log level to Warning

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?


```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
* Run webapp locally, see logs in stdout
* Run webapp in Azure, see logs in App Insights
